core/src/test/java/org/apache/iceberg/TestSchemaUpdate.java

```diff
@@ -1223,4 +1223,98 @@ public void testMoveBetweenStructsFails() {
                 .moveBefore("s2.x", "s1.a")
                 .apply());
   }
+
+  @Test
+  public void testAddIdentifierField() {
+    Schema newSchema = new SchemaUpdate(SCHEMA, SCHEMA_LAST_COLUMN_ID)
+        .allowIncompatibleChanges()
```

Comment:
I'd prefer to separate the existing column case from the new column case. And these tests should also validate that adding the identifier field before adding the column fails even if they are part of the same `SchemaUpdate` operation.